### PR TITLE
libinjector: add stack marker

### PR DESCRIPTION
**Pull request by skvl**
This allows to check some errors while injections

### CODE REVIEW

**Comment by skvl** [comment]
> Thank you @skvl, perhaps the marker could be something meaningful in ascii so it could be very useful for debugging or when viewing dumps.
> 
> For example: 0x6472616b76756624 (drakvuf$)
> 
> What do you think?

I don't hard code marker value. This allows in every plug-in use it's own value. E.g. I have used different values in `libusermode`, `fileextractor` and `procdump`.

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by skvl** [comment]
@tklengyel , is this OK?

File: src/libinjector/libinjector.h [hunk 1]
```diff
@@ -245,7 +245,8 @@ bool inject_function_call(
     x86_registers_t* regs,
     struct argument args[],
     int nb_args,
-    addr_t function_addr);
+    addr_t function_addr,
+    uint64_t* stack_marker);
```
Comment (tklengyel) [inline src/libinjector/libinjector.h:249]: Looks like `NOEXCEPT` is missing from this function, might as well add it now that you are modifying it.

**Comment by skvl** [comment]
I have re-based the branch

**Comment by tklengyel** [comment]
@drakvuf-jenkins Test this please

**Pull request merged**